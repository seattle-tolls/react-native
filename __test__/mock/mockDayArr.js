const mockDayArr = [
  {
    'goodToGo': '$2.70',
    'payByMail': '$4.70',
    'schedule': '10 a.m. to 1:59 p.m.',
    'dayName': 'Monday',
  },
  {
    'goodToGo': '$1.25',
    'payByMail': '$3.25',
    'schedule': 'Midnight to 4:59 a.m.',
    'dayName': 'Wednesday',
  },
  {
    'goodToGo': '$1.40',
    'payByMail':'$3.40',
    'schedule': '9 p.m. to 10:59 p.m.',
    'dayName': 'Saturday',
  },
  {
    'goodToGo': '$3.40',
    'payByMail': '$5.40',
    'schedule': '6 a.m. to 6:59 a.m.',
    'dayName': 'Wednesday',
    'holidayName': 'Christmas Day',
  },
]

export default mockDayArr