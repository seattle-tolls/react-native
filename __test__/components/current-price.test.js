import React from 'react'
import Adapter from 'enzyme-adapter-react-16'

import { shallow, configure } from 'enzyme'
import mockDayArr from '../mock/mockDayArr'

configure({ adapter: new Adapter() })

import CurrentPrice from '../../src/components/current-price'

describe('<CurrentPrice />', () => {
  mockDayArr.map(props => {
    const { goodToGo, payByMail, schedule, dayName, holidayName } = props

    let wrapper = shallow(<CurrentPrice { ...props }/>)
    it(`should render component for ${holidayName || dayName}`, () => {
      expect(wrapper.props()).toMatchSnapshot()
      expect(wrapper.find('Text').contains(goodToGo)).toBe(true)
      expect(wrapper.find('Text').contains(payByMail)).toBe(true)
      expect(wrapper.find('Text').contains(schedule)).toBe(true)
      expect(wrapper.find('Text').contains(dayName)).toBe(true)
      holidayName && expect(wrapper.find('Text').contains(holidayName)).toBe(true)
    })
  })
})

