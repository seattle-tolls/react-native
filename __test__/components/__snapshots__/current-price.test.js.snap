// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrentPrice /> should render component for Christmas Day 1`] = `
Object {
  "children": Array [
    <Text>
      Christmas Day
    </Text>,
    <Text>
      Wednesday
    </Text>,
    <Text>
      Good To Go
    </Text>,
    <Text>
      $3.40
    </Text>,
    <Text>
      Pay By Mail
    </Text>,
    <Text>
      $5.40
    </Text>,
    <Text>
      6 a.m. to 6:59 a.m.
    </Text>,
  ],
  "style": Object {
    "alignItems": "center",
    "backgroundColor": "yellow",
    "flex": 5,
    "justifyContent": "center",
    "width": "95%",
  },
}
`;

exports[`<CurrentPrice /> should render component for Monday 1`] = `
Object {
  "children": Array [
    undefined,
    <Text>
      Monday
    </Text>,
    <Text>
      Good To Go
    </Text>,
    <Text>
      $2.70
    </Text>,
    <Text>
      Pay By Mail
    </Text>,
    <Text>
      $4.70
    </Text>,
    <Text>
      10 a.m. to 1:59 p.m.
    </Text>,
  ],
  "style": Object {
    "alignItems": "center",
    "backgroundColor": "yellow",
    "flex": 5,
    "justifyContent": "center",
    "width": "95%",
  },
}
`;

exports[`<CurrentPrice /> should render component for Saturday 1`] = `
Object {
  "children": Array [
    undefined,
    <Text>
      Saturday
    </Text>,
    <Text>
      Good To Go
    </Text>,
    <Text>
      $1.40
    </Text>,
    <Text>
      Pay By Mail
    </Text>,
    <Text>
      $3.40
    </Text>,
    <Text>
      9 p.m. to 10:59 p.m.
    </Text>,
  ],
  "style": Object {
    "alignItems": "center",
    "backgroundColor": "yellow",
    "flex": 5,
    "justifyContent": "center",
    "width": "95%",
  },
}
`;

exports[`<CurrentPrice /> should render component for Wednesday 1`] = `
Object {
  "children": Array [
    undefined,
    <Text>
      Wednesday
    </Text>,
    <Text>
      Good To Go
    </Text>,
    <Text>
      $1.25
    </Text>,
    <Text>
      Pay By Mail
    </Text>,
    <Text>
      $3.25
    </Text>,
    <Text>
      Midnight to 4:59 a.m.
    </Text>,
  ],
  "style": Object {
    "alignItems": "center",
    "backgroundColor": "yellow",
    "flex": 5,
    "justifyContent": "center",
    "width": "95%",
  },
}
`;
